### PR TITLE
Fix priorities being incorrect in the task list.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/66292.

### CODE REVIEW

File: src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs [hunk 1]
```diff
@@ -34,8 +41,15 @@ public static ImmutableArray<TaskListItemDescriptor> Parse(ImmutableArray<string
...
                 if (item.Split(':') is [var token, var priorityString] &&
                     !string.IsNullOrWhiteSpace(token) &&
-                    int.TryParse(priorityString, NumberStyles.None, CultureInfo.InvariantCulture, out var priority))
+                    int.TryParse(priorityString, NumberStyles.None, CultureInfo.InvariantCulture, out var integer))
                 {
+                    var priority = integer switch
+                    {
+                        1 => TaskListItemPriority.Low,
+                        2 => TaskListItemPriority.Medium,
+                        3 => TaskListItemPriority.High,
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: Why is this one 1-3 instead of 0-2?

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: it's just how the editor stores and sends this stuff to us.  @olegtk are you aware of any reason for that?  or it just just purely historical?

Comment (olegtk) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: not sure. which editor component you are getting this stuff from?

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: @olegtk This is the value read out of the roaming-storage location: `Microsoft.VisualStudio.ErrorListPkg.Shims.TaskListOptions.CommentTokens`.

Comment (sandyarmstrong) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: That's interesting, I think on VSMac we've been sending 0 for low and 2 for high...

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: 😢 .  oh no...

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: @sandyarmstrong great!

Comment (olegtk) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: LOL, I don't think anybody remembers why:
```
// I've no idea why the strange conversion (legacy mapping from __VSERRORCATEGORY?).
private static int EncodedValueFromPriority(CommentTaskPriority p) { return 3 - (int)p; }
```

https://devdiv.visualstudio.com/DevDiv/_git/VS?path=/src/env/ErrorList/Pkg/Shims/TaskListOptions.cs&version=GBmain&line=133&lineEnd=134&lineStartColumn=1&lineEndColumn=98&lineStyle=plain&_a=contents

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:50]: no worries.  it is what it is :)  

I'll add a comment to this effect.  :)

**Comment by CyrusNajmabadi** [comment]
@jmarolf @akhera99  ptal. Thanks!

File: src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs [hunk 2]
```diff
@@ -34,8 +41,25 @@ public static ImmutableArray<TaskListItemDescriptor> Parse(ImmutableArray<string
             {
                 if (item.Split(':') is [var token, var priorityString] &&
                     !string.IsNullOrWhiteSpace(token) &&
-                    int.TryParse(priorityString, NumberStyles.None, CultureInfo.InvariantCulture, out var priority))
+                    int.TryParse(priorityString, NumberStyles.None, CultureInfo.InvariantCulture, out var encoded))
                 {
+                    // From:
+                    // https://devdiv.visualstudio.com/DevDiv/_git/VS?path=/src/env/ErrorList/Pkg/Shims/TaskListOptions.cs&version=GBmain&line=133&lineEnd=134&lineStartColumn=1&lineEndColumn=98&lineStyle=plain&_a=contents
+#if false
```
Comment (dibarbet) [inline src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:48]: wonder if we should put this in a documentation comment so we can put it in a code block or something

**Review by dibarbet — APPROVED** [review]

File: src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/TaskList/TaskListItemDescriptor.cs:48]: i'm fine any way :)

**Pull request merged**